common/test/acceptance/tests/lms/test_problem_types.py

```diff
@@ -381,12 +388,85 @@ def test_partially_complete_notifications(self):
         self.problem_page.wait_partial_notification()
 
 
-class AnnotationProblemTypeTest(ProblemTypeTestBase, ProblemTypeTestMixin):
+@ddt.ddt
+class ProblemNeverShowCorrectnessMixin(object):
+    """
+    Tests the effect of adding `show_correctness: never` to the sequence metadata
+    for subclasses of ProblemTypeTestMixin.
+    """
+    sequential_metadata = {'show_correctness': 'never'}
+
+    @attr(shard=7)
+    @ddt.data('correct', 'incorrect', 'partially-correct')
+    def test_answer_says_submitted(self, correctness):
+        """
+        Scenario: I can answer a problem <Correctness>ly
+        Given External graders respond "<Correctness>""
+        And I am viewing a "<ProblemType>" problem
+        in a subsection with show_correctness set to "never"
+        Then I should see a score of "N point(s) possible (ungraded, results hidden)"
+        When I answer a "<ProblemType>" problem "<Correctness>ly"
+        Then my "<ProblemType>" answer is marked "<Correctness>"
+        And The "<ProblemType>" problem displays only a "submitted"
+        notification.
+        And I should see a score of "N point(s) possible (ungraded, results hidden)"
+        And a "problem_check" server event is emitted
+        And a "problem_check" browser event is emitted
+        """
+
+        # Not all problems have partially correct solutions configured
+        if correctness == 'partially-correct' and not self.partially_correct:
+            raise SkipTest("Test incompatible with the current problem type")
+
+        # Problem progress text depends on points possible
+        possible = 'possible (ungraded, results hidden)'
+        if self.problem_points == 1:
+            problem_progress = '1 point {}'.format(possible)
+        else:
+            problem_progress = '{} points {}'.format(self.problem_points, possible)
+
+        # Make sure we're looking at the right problem
+        self.problem_page.wait_for(
+            lambda: self.problem_page.problem_name == self.problem_name,
+            "Make sure the correct problem is on the page"
+        )
+
+        # Learner can see that score will be hidden prior to submitting answer
+        self.assertEqual(self.problem_page.problem_progress_graded_value, problem_progress)
```

Comment:
A concern about this is that it will be (more) challenging for a learner to know that they have in fact submitted an answer. In other cases the problem progress is different before a submission vs. after a submission. There is the text indicating the number of remaining attempts, for problems that have a finite number of attempts (presumably that is most problems).